experimental/packages/opentelemetry-instrumentation/src/semver.ts

```diff
@@ -0,0 +1,585 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
A comment near the top of the file giving the basic justification for this semver.ts (i.e. that it is about startup time vs. node-semver, limited to just `satisfies()`, etc.) would be helpful for future maintainers.

Reply:
done